test/integration/servicedns_test.go

```diff
@@ -204,3 +196,24 @@ func newServiceDNSTestFixture(tl common.TestLogger, fedFixture *framework.Federa
 
 	return f
 }
+
+func updateFederatedClusterStatus(tl common.TestLogger, client fedclientset.Interface, clusterName string, clusterStatus fedv1a1.FederatedClusterStatus) {
+	err := wait.PollImmediate(framework.DefaultWaitInterval, wait.ForeverTestTimeout, func() (exist bool, err error) {
+		federatedCluster, err := client.CoreV1alpha1().FederatedClusters(util.FederationSystemNamespace).Get(clusterName, metav1.GetOptions{})
+		if err != nil {
+			tl.Logf("Error retrieving federated cluster %q: %v", clusterName, err)
+			return false, err
+		}
+		federatedCluster.Status.Region = clusterStatus.Region
+		federatedCluster.Status.Zone = clusterStatus.Zone
```

Comment:
update fields individually, so that existing cluster status conditions are not overwritten.

Reply:
Consider adding this as a comment so it will be available to future maintainers.

Reply:
Makes sense. Added a comment. Thanks !